aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -316,6 +361,10 @@ inline void Dispatcher::callBoxed(const OperatorHandle& op, Stack* stack) const
   const auto& dispatchTable = op.operatorIterator_->op.dispatch_table();
   auto dispatchKey = dispatchTable.dispatchKeyExtractor().getDispatchKeyBoxed(backendsWithoutFallthrough_, stack);
   const KernelFunction& kernel = dispatch_(dispatchTable, dispatchKey);
+
+  // using already existing stack to record function execution in observers
+  RECORD_FUNCTION(op.schema().name(), *stack, at::FunctionSequenceNumber::peek());
```

Comment:
if we do it manually in `callUnboxed`, I'd rather do it manually here doo. It's weird that one location uses macros and the other one doesn't. You can probably factor out common code between the record_function implementation in `callUnboxed` and here into subfunctions.

Reply:
makes sense